### PR TITLE
Mathieu/debug elimination

### PR DESCRIPTION
**Pull request by MathieuDutSik**
# Pull Request Summary
 
Elimination of many print statement that are just useless and plainly annoying.

## Description

It cleans up the code which makes it clearer.

### Issue(s) addressed

There was a number of discussions on the merits of #ifdef statements and the consensus was that we should have less of them.

### Commit Message

Reduction in the number of print statements.

### Check list  

- [ ] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [ ] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [ ] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [ ] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing

* How were these changes tested?
* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)
* Have the matrix regression tests been run (if yes, please note HPC and compiler)?
* Please indicate the expected changes in the regression test output, (Note the [list](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-develop#4-look-at-results) of known non-identical tests.)
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):

### CODE REVIEW

File: model/src/w3fldsmd.F90 [hunk 1]
```diff
@@ -200,9 +200,7 @@ SUBROUTINE W3FLDO ( INXOUT, IDFLD, NDS, NDST, NDSE, NX, NY,     &
       USE W3SERVMD, ONLY: STRACE
 #endif
 !
-#ifdef W3_DEBUGFLS
       USE W3ODATMD, only : IAPROC
```
Comment (JessicaMeixner-NOAA) [inline model/src/w3fldsmd.F90:203]: Can we remove this use statement as well or keep the if-def with this use statement if it's needed elsewhere?

File: model/src/ww3_ounf.F90 [hunk 2]
```diff
@@ -169,9 +169,7 @@ PROGRAM W3OUNF
       USE W3IOGOMD, ONLY: W3IOGO, W3READFLGRD, W3FLGRDFLAG
       USE W3INITMD, ONLY: WWVER, SWITCHES
       USE W3ODATMD, ONLY: NAPROC, NOSWLL, PTMETH, PTFCUT
-#ifdef W3_DEBUG
       USE W3ODATMD, only : IAPROC
```
Comment (JessicaMeixner-NOAA) [inline model/src/ww3_ounf.F90:172]: I think this line should either be reomved or remain behind W3_DEBUG.

**Review by JessicaMeixner-NOAA — COMMENTED** [review]
@MathieuDutSik two minor questions/comments.  I have not yet run regression tests, but I'd be happy to help with that process.  We can also run with a few of the remaining DEBUG options if needed to ensure everything is still working.  

It'd be great to hear what the path for this getting back to NOAA-EMC/WW3 is as well, so everyone can appreciate your work here! Thanks again @MathieuDutSik !

File: model/src/ww3_ounf.F90 (hunk 2 above)
Comment (MathieuDutSik) [reply model/src/ww3_ounf.F90:172]: Thanks done.

File: model/src/w3fldsmd.F90 (hunk 1 above)
Comment (MathieuDutSik) [reply model/src/w3fldsmd.F90:203]: Thanks, done.

**Comment by JessicaMeixner-NOAA** [comment]
Thanks for the updates @MathieuDutSik  -- I approve of this PR assuming it passes all the regtests (haven't run them myself so not putting approve yet).  @thesser1 what's the path forward for this PR to get into noaa-emc/ww3 develop branch? It'd be great to do that ASAP!

File: model/src/ww3_ounf.F90 (hunk 2 above)
Comment (JessicaMeixner-NOAA) [reply model/src/ww3_ounf.F90:172]: Did your updates get pushed to your branch? I don't see updates on the code here.

**Comment by MathieuDutSik** [comment]
Thank you for your kind words. It is indeed necessary to pass the regtests as avoiding the regtests should be done only in very special cases.
I have further corrections to WW3 down the line after this one, though not as significant.

**Review by JessicaMeixner-NOAA — APPROVED** [review]
@thesser1 what's the plan for getting this to noaa-emc/ww3 develop branch?

**Comment by thesser1** [comment]
[matrixCompFull.txt](https://github.com/erdc/WW3/files/9602052/matrixCompFull.txt)
[matrixCompSummary.txt](https://github.com/erdc/WW3/files/9602056/matrixCompSummary.txt)
[matrixDiff.txt](https://github.com/erdc/WW3/files/9602057/matrixDiff.txt)
All regtests looked good on my end.

**Pull request merged**